packages/starknet-snap/src/signTransaction.ts

```diff
@@ -0,0 +1,27 @@
+import { Signature, Signer, stark } from 'starknet';
+import { ApiParams, SignTransactionParams } from './types/snapApi';
+import { getKeysFromAddress } from './utils/starknetUtils';
+import { getNetworkFromChainId } from './utils/snapUtils';
+
+export async function signTransaction(params: ApiParams): Promise<Signature> {
+  const { state, keyDeriver, requestParams } = params;
+  const requestParamsObj = requestParams as SignTransactionParams;
+  const userAddress = requestParamsObj.userAddress;
+  const network = getNetworkFromChainId(state, requestParamsObj.chainId);
+  const { privateKey } = await getKeysFromAddress(keyDeriver, network, state, userAddress);
```

Comment:
account should check if it has deployed

Reply:
Do we need to? Because this seems to be just for signing a transaction.

If it is from a UX perspective, then it makes sense!

Reply:
TBC